apollo-federation/src/link/federation_spec_definition.rs

```diff
@@ -539,6 +553,178 @@ impl FederationSpecDefinition {
+        )
+    }
+
+    fn fields_argument_specification() -> DirectiveArgumentSpecification {
+        DirectiveArgumentSpecification {
+            base_spec: ArgumentSpecification {
+                name: FEDERATION_FIELDS_ARGUMENT_NAME,
+                get_type: |_| Ok(Type::Named(FIELDSET_SCALAR_NAME)),
+                default_value: None,
+            },
+            composition_strategy: None,
+        }
+    }
+
+    fn resolvable_argument_specification() -> DirectiveArgumentSpecification {
+        DirectiveArgumentSpecification {
+            base_spec: ArgumentSpecification {
+                name: FEDERATION_RESOLVABLE_ARGUMENT_NAME,
+                get_type: |_| Ok(ty!(Boolean)),
+                default_value: Some(Value::Boolean(true)),
+            },
+            composition_strategy: None,
+        }
+    }
+
+    fn requires_directive_specification() -> DirectiveSpecification {
+        DirectiveSpecification::new(
+            FEDERATION_REQUIRES_DIRECTIVE_NAME_IN_SPEC,
+            &[Self::fields_argument_specification()],
+            false,
+            &[DirectiveLocation::FieldDefinition],
+            false,
+            None,
+        )
+    }
+
+    fn provides_directive_specification() -> DirectiveSpecification {
+        DirectiveSpecification::new(
+            FEDERATION_PROVIDES_DIRECTIVE_NAME_IN_SPEC,
+            &[Self::fields_argument_specification()],
+            false,
+            &[DirectiveLocation::FieldDefinition],
+            false,
+            None,
+        )
+    }
+
+    fn external_directive_specification() -> DirectiveSpecification {
+        DirectiveSpecification::new(
+            FEDERATION_EXTERNAL_DIRECTIVE_NAME_IN_SPEC,
+            &[DirectiveArgumentSpecification {
+                base_spec: ArgumentSpecification {
+                    name: FEDERATION_REASON_ARGUMENT_NAME,
+                    get_type: |_| Ok(ty!(String)),
+                    default_value: None,
+                },
+                composition_strategy: None,
+            }],
+            false,
+            &[DirectiveLocation::FieldDefinition],
+            false,
+            None,
+        )
+    }
+
+    fn extends_directive_specification() -> DirectiveSpecification {
+        DirectiveSpecification::new(
+            FEDERATION_EXTENDS_DIRECTIVE_NAME_IN_SPEC,
+            &[],
+            false,
+            &[DirectiveLocation::Object, DirectiveLocation::Interface],
+            false,
+            None,
+        )
+    }
+
+    fn shareable_directive_specification(&self) -> DirectiveSpecification {
+        DirectiveSpecification::new(
+            FEDERATION_SHAREABLE_DIRECTIVE_NAME_IN_SPEC,
+            &[],
+            self.version().gt(&Version { major: 2, minor: 1 }), // repeatable in Fed 2.2 or later
+            &[DirectiveLocation::FieldDefinition],
+            false,
+            None,
+        )
+    }
+
+    fn override_directive_specification() -> DirectiveSpecification {
```

Comment:
This seems to be the pre-2.7 version of override. It seems worth commenting about that.